### ISSUE TITLE
Wrong Error for item assignment to a list element

### ISSUE THREAD

**Issue by purplesun84** [PyCQA/pylint#2160]
### Steps to reproduce
Applied pylint on the following code:
```
class Test():

    def __init__(self):
        self.param = {'A':'', 'B':''}
        return

    def AssignValue(self):
        self.param['A'] = ['Test', '1', 2, None]
        return

    def ReproduceError(self):
        self.param['A'][0] += '/Testadd'
        return


if __name__ == "__main__":
    TestInstance = Test()
    TestInstance.AssignValue()
    TestInstance.ReproduceError()`
```

### Current behavior
...
E: 12, 8: "self.param['A']" does not support item assignment (unsupported-assignment-operation)
...

### Expected behavior
No Error

### pylint --version output
pylint 1.8.2, 
astroid 1.6.1
Python 2.7.14 |Anaconda, Inc.| (default, Dec  7 2017, 17:05:42) 
[GCC 7.2.0]

**Comment by PCManticore** [comment]
You get this error because pylint is not smart enough to figure out the data flow from your program. From its perspective, the value that you're trying to assign to at line 12 is a string, not a dictionary. One way to get rid of the error is to initialize the param list with empty lists instead of empty strings.
